models/DataObject/Service.php

```diff
@@ -1815,11 +1818,17 @@ protected static function mapFieldname(string $field, array $helperDefinitions):
                 $groupConfig = DataObject\Classificationstore\GroupConfig::getById($groupId);
                 $keyConfig = DataObject\Classificationstore\KeyConfig::getById($keyId);
 
-                $field = $fieldname . '~' . $groupConfig->getName() . '~' . $keyConfig->getName();
+                $key = $fieldname . '~' . $groupConfig->getName() . '~' . $keyConfig->getName();
             }
         }
 
-        return $field;
+        if ($header === 'name') {
+            return $key;
+        } elseif ($header === 'no_header') {
+            return '';
+        }
```

Comment:
Maybe this part can be moved to the top of the method? Because it seems that it could be an early return with no need for further processing.

```php
if ($header === 'no_header') {
    return '';
}
```

Reply:
Indeed, thank you for pointing that out